sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.net6.0.cs

```diff
@@ -110,86 +110,80 @@ namespace Azure.Core.Dynamic
             object System.Collections.IEnumerator.Current { get { throw null; } }
             public void Dispose() { }
-            public Azure.Core.Dynamic.DynamicJson.ArrayEnumerator GetEnumerator() { throw null; }
+            public Azure.Core.Dynamic.DynamicData.ArrayEnumerator GetEnumerator() { throw null; }
             public bool MoveNext() { throw null; }
             public void Reset() { }
-            System.Collections.Generic.IEnumerator<Azure.Core.Dynamic.DynamicJson> System.Collections.Generic.IEnumerable<Azure.Core.Dynamic.DynamicJson>.GetEnumerator() { throw null; }
+            System.Collections.Generic.IEnumerator<Azure.Core.Dynamic.DynamicData> System.Collections.Generic.IEnumerable<Azure.Core.Dynamic.DynamicData>.GetEnumerator() { throw null; }
             System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
         }
         [System.Diagnostics.DebuggerDisplayAttribute("{Current,nq}")]
         [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
-        public partial struct ObjectEnumerator : System.Collections.Generic.IEnumerable<Azure.Core.Dynamic.DynamicJsonProperty>, System.Collections.Generic.IEnumerator<Azure.Core.Dynamic.DynamicJsonProperty>, System.Collections.IEnumerable, System.Collections.IEnumerator, System.IDisposable
+        public partial struct ObjectEnumerator : System.Collections.Generic.IEnumerable<Azure.Core.Dynamic.DynamicDataProperty>, System.Collections.Generic.IEnumerator<Azure.Core.Dynamic.DynamicDataProperty>, System.Collections.IEnumerable, System.Collections.IEnumerator, System.IDisposable
         {
             private object _dummy;
             private int _dummyPrimitive;
-            public Azure.Core.Dynamic.DynamicJsonProperty Current { get { throw null; } }
+            public Azure.Core.Dynamic.DynamicDataProperty Current { get { throw null; } }
             object System.Collections.IEnumerator.Current { get { throw null; } }
             public void Dispose() { }
-            public Azure.Core.Dynamic.DynamicJson.ObjectEnumerator GetEnumerator() { throw null; }
+            public Azure.Core.Dynamic.DynamicData.ObjectEnumerator GetEnumerator() { throw null; }
             public bool MoveNext() { throw null; }
             public void Reset() { }
-            System.Collections.Generic.IEnumerator<Azure.Core.Dynamic.DynamicJsonProperty> System.Collections.Generic.IEnumerable<Azure.Core.Dynamic.DynamicJsonProperty>.GetEnumerator() { throw null; }
+            System.Collections.Generic.IEnumerator<Azure.Core.Dynamic.DynamicDataProperty> System.Collections.Generic.IEnumerable<Azure.Core.Dynamic.DynamicDataProperty>.GetEnumerator() { throw null; }
             System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
         }
     }
-    public enum DynamicJsonNameMapping
+    public enum DynamicDataNameMapping
```

Comment:
I like keeping this one general because I think it would apply to any subtype/extension type.  Is there a format we might consider where values aren't keyed on property names?